### PR TITLE
fix: anchor links types and type-safety

### LINKED ISSUE #213: Type checking is broken in the 0.7.1 release
See here: https://github.com/nuxt-modules/mdc/pull/206#issuecomment-2131374361

The following recent PRs have broken type checking in downstream projects:

- https://github.com/nuxt-modules/mdc/pull/180
- https://github.com/nuxt-modules/mdc/pull/206

Would it be possible to add type checking in a simple Nuxt app to the test pipeline in the repo?

### PR DESCRIPTION
**Pull request by adamdehaven**
Updates prose heading components to resolve #208 and #206. Add type checking in the pipeline to resolve #213.

The current `0.7.1` release is unusable in host apps with type checking enabled.

---

@stefanobartoletti I integrated your changes from #208 for the `boolean` and interface so everything in downstream apps should work out of the box.

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "@nuxtjs/plausible": "^1.0.0",
     "@types/node": "^20.12.12",
     "eslint": "^9.2.0",
-    "nuxt": "^3.11.2"
+    "nuxt": "^3.11.2",
+    "pinceau": "0.18.9"
```
Comment (adamdehaven) [inline docs/package.json:22]: `pinceau` is imported but not explicitly declared as a devDependency; resolves an import error

File: nuxt.config.ts [hunk 2]
```diff
@@ -9,7 +9,12 @@ export default defineNuxtConfig({
             fileURLToPath(new URL('./src/config.ts', import.meta.url))
           ]
         }
-      }
-    }
+      },
+      exclude: [
+        fileURLToPath(new URL('./docs', import.meta.url))
+      ]
```
Comment (adamdehaven) [inline nuxt.config.ts:15]: Exclude type checking the local `./docs/` directory; doesn't seem to resolve everything correctly and doesn't impact the actual module release.

File: test/fixtures/basic/pages/component-map/vue-component.vue [hunk 3]
```diff
@@ -12,7 +12,7 @@
 </template>
 
 <script setup lang="ts">
-import CustomParagraph from '~/components/CustomParagraph.vue'
+import CustomParagraph from '../../components/CustomParagraph.vue'
```
Comment (adamdehaven) [inline test/fixtures/basic/pages/component-map/vue-component.vue:15]: Alias was not resolving properly with type checking enabled.

**Comment by stefanobartoletti** [comment]
@adamdehaven I think it is fine by me, as long as it retains the ability to set or disable all links at once without necessarily using granular options.

Anyway, the main author is the person to check with :)

File: playground/nuxt.config.ts [hunk 4]
```diff
@@ -3,6 +3,7 @@ export default defineNuxtConfig({
     '@nuxt/ui',
     '../src/module'
   ],
+  // @ts-expect-error - Ignore invalid property error; mdc is a local module
```
Comment (danielroe) [inline playground/nuxt.config.ts:6]: I feel this shouldn’t be needed. I think playground should be excluded in root tsconfig and playground type checking should be run separately.

Comment (adamdehaven) [reply playground/nuxt.config.ts]: @danielroe updated 👍🏼

**Review by farnabaz — APPROVED** [review]
LGTM ❤️

**Pull request merged**